packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -161,6 +192,31 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
         };
         updateFocusContextKeys();
         this.shell.activeChanged.connect(updateFocusContextKeys);
+        this.workspaceCommandContribution.onDidCreateNewFile(async event => this.onDidCreateNewResource(event));
+        this.workspaceCommandContribution.onDidCreateNewFolder(async event => this.onDidCreateNewResource(event));
+    }
+
+    private async onDidCreateNewResource(event: DidCreateNewResourceEvent): Promise<void> {
+        const navigator = this.tryGetWidget();
+        if (!navigator || !navigator.isVisible) {
+            return;
+        }
+        const model: FileNavigatorModel = navigator.model;
+        for (const uri of event.uri.allLocations) {
```

Comment:
why do we need it? we have `event.parent` already? just:
```ts
const parent = await model.revealFile(event.parent);
if (DirNode.is(event.parent)) {
    await model.refresh(event.parent);
}
```

Reply:
fixed